cmake/CPM.cmake

```diff
@@ -773,10 +834,32 @@ function(cpm_parse_option OPTION)
       "${OPTION_KEY}"
       PARENT_SCOPE
   )
-  set(OPTION_VALUE
-      "${OPTION_VALUE}"
-      PARENT_SCOPE
+  # https://cmake.org/cmake/help/latest/command/if.html#basic-expressions (1/0 is problematic)
```

Comment:
why is 1/0 problematic?

Reply:
It could mean `"ON"` or `NUMBER_OF_CORE "1"`

Reply:
Then why change all values to "1"? Instead of to "ON". Thus this case will be covered

Reply:
@TheLartians yes good point, this it's an other problem. Maybe CPM could accept only TRUE YES ON. It makes it more informative and I'm  temptated to say using 1 or Y for TRUE should be avoided.  `I think we should either use the normalisation only when checking of parameter consistency or add a note in the readme that we recommend truthy/falsy values to be standardised in CPM.cmake options (e.g. to YES/NO).` I agree totlaly to this point but maybe we could add TRUE and On as they are not preblematic and very well used. I have never seen 1 or Y.